### DISCUSSION TITLE
Fly & god improvements

### DISCUSSION THREAD

**Discussion by Rukes** [Zrips/CMI#971] [Enhancement]
1. If a player has enabled fly (he has permission too) and disconnects, flight mode gets disabled. God remains enabled after disconnect and next connection.
2. automatically disable fly or god when a player joins, if he does not have permission for it
3. does fly and god support silent `-s` variable? if no please add it 🐱

**Comment by Kryniowesegryderiusz** [comment]
It would be better to have this in config. I would love to have flight mode still enabled after login, ig someone still have permission to it <3

**Comment by Rukes** [comment]
Yes, that's it. I mean to remain enabled fly after relog, bcs it gets disabled when you disconnect. Otherwise, if you don't have the permission to fly (or god) it will be disabled.

**Comment by LogGits** [comment]
Im pretty sure 2 and 3 already work as you mentioned.

**Comment by TheoRetisch1** [comment]
I'm also in favor of improving the feature. If you are in gm0 you always lose flight mode after a relog regardless of permission and config settings. It would be great if the permission could be kept after a relog!

**Comment by ghost** [comment]
To concretize the matter:

- Gamemode 0: No matter what permissions you have, the fly mode is always removed. 
You will fall down or get set to the ground depending on your config settings.

- Gamemode 1: The flight mode is not removed after a relog, but it is suspended. 
So you fall until you press the spacebar.


It would be great if you could save the flight mode, so that one just float on the spot in the air after a relog.

CMI: 8.3.10.0
Paper 1.13.2 #506

**Comment by Kryniowesegryderiusz** [comment]
So what about that?

**Comment by APG77** [comment]
Fly mode is still removed when players relog in survival. I've thoroughly searched the config file and there is still no option to fix this.

CMI version 8.6.19.5
Paper 1.15.2 build 112

**Comment by Kryniowesegryderiusz** [comment]
Still happens
